File: chat-backend/src/main/java/com/smallchat/backend/user/framework/jpa_adapter/UserRepository.java
package com.smallchat.backend.user.framework.jpa_adapter;

import com.smallchat.backend.user.domain.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, String> {

    Optional<User> findByLoginId(String loginId);

    Boolean existsByLoginId(String loginId);
}

File: chat-backend/src/main/java/com/smallchat/backend/user/domain/interfaces/UserRepository.kt
package com.smallchat.backend.user.domain.interfaces

import com.smallchat.backend.user.domain.model.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, String> {
    fun findByLoginId(loginId: String): User?
    fun isExistsByLoginId(loginId: String?): Boolean
}
